source/lib.ts

```diff
@@ -144,13 +144,24 @@ const parseOptions = (passedOptions: Partial<Options>): Configuration => {
 
 			return request.ip
 		},
-		handler(
+		async handler(
 			_request: Request,
 			response: Response,
 			_next: NextFunction,
 			_optionsUsed: Options,
-		): void {
-			response.status(config.statusCode).send(config.message)
+		): Promise<void> {
+			let { message } = config as { message: unknown }
+			// Set the response status code
+			response.status(config.statusCode)
+			// Call the `message` if it is a function.
+			if (typeof message === 'function') {
+				message = await message(_request, response)
+			}
+
+			// Send the response if headers are not sent.
+			if (!response.headersSent) {
```

Comment:
Could you please use `!response.writableEnded` here instead?

Reply:
I figured there was something better than `headersSent`, Thank you!